source/modulo_components/test/cpp/test_component_interface_parameters.cpp

```diff
@@ -67,6 +69,22 @@ TYPED_TEST(ComponentInterfaceParameterTest, AddNameValueParameter) {
   this->template expect_parameter_value<int>(1);
 }
 
+TYPED_TEST(ComponentInterfaceParameterTest, AddEmtpyParameter) {
+  EXPECT_THROW(auto discard = this->component_->get_parameter("test"), exceptions::ComponentParameterException);
+  EXPECT_THROW(this->component_->get_ros_parameter("test"), rclcpp::exceptions::ParameterNotDeclaredException);
+  this->component_->template add_parameter<int>("test", "Test parameter");
+
+  // Adding the parameter should declare and set the value and call the validation function
+  EXPECT_TRUE(this->component_->validate_parameter_was_called);
+  EXPECT_NO_THROW(auto discard = this->component_->get_parameter("test"));
+  EXPECT_NO_THROW(this->component_->get_ros_parameter("test"));
+  EXPECT_STREQ(this->component_->get_ros_parameter("test").value_to_string().c_str(), "not set");
+  EXPECT_THROW(this->component_->template get_parameter_value<int>("test"),
```

Comment:
Should this maybe throw a ComponentParameterError

Reply:
Wait now I'm a bit confused. You want to add an `EmptyParameterException` to state representation?

I was suggesting throwing a `ComponentParameterException` (already exists) such that any component exception can be caught with the base exception class `ComponentException` but I agree with you that we should change anything here.

Reply:
I was mainly agreeing that accessing an empty parameter is not the component's fault, and so should not be a `ComponentException` or `ComponentParameterException`. Rather it's a state representation exception, so catching just to rethrow at component scope doesn't make sense imo. 

But I just added the related thought that, if you wanted to be more specific about the emptiness, another exception class in state representation could potentially be nice. It's not needed and changes nothing except making empty parameters maybe a little friendlier to catch

Reply:
I see yes, thanks